packages/material-icon/index.js

```diff
@@ -2,34 +2,80 @@ import React from 'react';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
 
+import withRipple from '@material/react-ripple';
+
 export default class MaterialIcon extends React.Component {
   render() {
     const {
       icon,
-      className,
+      hasRipple,
       ...otherProps
     } = this.props;
-    const classes = classnames('material-icons', {
-      [className]: className.length > 0,
-    });
+
+    if (hasRipple) {
```

Comment:
so this is happening in the render method...does that mean the entire MaterialIcon component will re-render *with* a ripple if someone changes the `hasRipple` prop to true at runtime?

Reply:
That's correct. It will be re-rendered, but that's ok since the underlying HTML will not change. Also in any other case we would expect it to re-render since its a new component. 

With React we shouldn't worry about re-rendering, since every time `props` or `state` changes, it goes through a render cycle. There are of course edge case, but this is what React was built for.